File: q0219/q219.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        index_map = {}
        for i in range(len(nums)):
            num = nums[i]
            if num in index_map:
                j = index_map[num]
                if i - j <= k: 
                    return True
            index_map[num] = i
        return False

File: q0219/q219.java
package q0219;
import java.util.*;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> indexMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (indexMap.containsKey(num)) {
                int j = indexMap.get(num);
                if (i - j <= k)
                    return true;
            }
            indexMap.put(num, i);
        }
        return false;
    }
}